### PR TITLE
Remove express when not required 

### PR DESCRIPTION
**Pull request by tobiasqueck**
I noticed that `axios-extension` is using `express` for a very simple server use case. I replaced it with usage of the `http` module and can reduce the installation size of the package and all its dependencies from 13m to 11m.
I also noticed that our middlewares are using a specific version of express and when they are used somewhere else by e.g. the ui5-tooling using a different express version, then express gets installed twice, so I replaced it as well with the simple `http` module

- [x] remove express from `axios-extension` (11mb -> 9mb installation size) 
- [x] remove express from `backend-proxy-middleware` (22mb -> 19mb installation size)
- [x] remove express from `ui5-proxy-middleware` (17mb -> 15mb installation size)

### CODE REVIEW

**Comment by tobiasqueck** [comment]
@ullasholla please closely look at the changes to the `axios-extension` to make sure I didn't change any logic. @zdravko-georgiev please do the same for the middlewares.
All unit tests run without issues and my manual tests also didn't show a problem but still, please look very closely.

**Comment by zdravko-georgiev** [comment]
> @ullasholla please closely look at the changes to the `axios-extension` to make sure I didn't change any logic (please also checkout the new unit test)
> 
> @zdravko-georgiev please do the same for the middlewares - especially the `ui5-proxy-middleware` because I also reduced the cognitive complexity of some of the code. All unit tests run without issues and my manual tests also didn't show a problem but still, please look very closely.

Thanks @tobiasqueck, I will check

**Review by zdravko-georgiev — DISMISSED** [review]
Code looks good.
Didn't found any issues when testing manually

**Review by zdravko-georgiev — DISMISSED** [review]

File: packages/axios-extension/src/auth/reentrance-ticket/redirect.ts [hunk 1]
```diff
@@ -30,35 +29,38 @@ export interface SetupRedirectOptions {
...
-        } else {
-            logger.error('Error getting reentrance ticket');
-            logger.debug(req);
-            res.set('Content-Type', 'text/html');
-            res.status(500);
-            res.send(Buffer.from(redirectErrorHtml()));
-            server.close();
-            reject(new ConnectionError('Error getting reentrance ticket'));
+    server = http.createServer((req, res) => {
+        const reqUrl = new URL(req.url, `http:${req.headers.host}`);
```
Comment (ullasholla) [inline packages/axios-extension/src/auth/reentrance-ticket/redirect.ts:44]: The second parameter to the `URL()` constructor gets interpolated to a value like `http:localhost:30034/redirect?...`

The code works but the URL standard specifies that the [absolute URL string](https://url.spec.whatwg.org/#absolute-url-string) must have `//` after the scheme. Consider changing this?

File: packages/axios-extension/src/auth/uaa.ts [hunk 2]
```diff
@@ -179,24 +178,27 @@ export class Uaa {
...
             const timer = setTimeout(handleTimeout, timeout);
-            app.get(Redirect.path, (req, res) => {
-                res.set('Content-Type', 'text/html');
-                res.send(Buffer.from(redirectSuccessHtml(this.logoutUrl, this.systemId)));
-                this.log.info('Got authCode');
-                resolve({ authCode: req.query.code + '', redirect });
-                if (timer) {
-                    clearTimeout(timer);
+            server = http.createServer((req, res) => {
+                const reqUrl = new URL(req.url, `http:${req.headers.host}`);
```
Comment (ullasholla) [inline packages/axios-extension/src/auth/uaa.ts:191]: Question about the interpolated base URL. Same as: https://github.com/SAP/open-ux-tools/pull/540/files#r974651046

**Review by ullasholla — DISMISSED** [review]
Thanks @tobiasqueck, looks good.
Have two minor queries. Could you take a look please?

File: packages/axios-extension/src/auth/uaa.ts (hunk 2 above)
Comment (tobiasqueck) [reply packages/axios-extension/src/auth/uaa.ts]: https://github.com/SAP/open-ux-tools/pull/540/commits/6eca7ee931b046459cc6f27a29d3d4ea4d99d050

File: packages/axios-extension/src/auth/reentrance-ticket/redirect.ts (hunk 1 above)
Comment (tobiasqueck) [reply packages/axios-extension/src/auth/reentrance-ticket/redirect.ts]: https://github.com/SAP/open-ux-tools/pull/540/commits/6eca7ee931b046459cc6f27a29d3d4ea4d99d050

**Review by ullasholla — APPROVED** [review]
@tobiasqueck, thanks for addressing the review comments. The changes and tests look good to me

**Comment by tobiasqueck** [comment]
I admin-merged it because @zdravko-georgiev is on vacation, however, he approved the relevant parts for him a few times already.

**Pull request merged**